### ISSUE TITLE
texture memory raises `cupy_backends.cuda.api.runtime.CUDARuntimeError: cudaErrorInvalidValue: invalid argument`

### ISSUE THREAD

**Issue by JoOkuma** [cupy/cupy#7980]
### Description

Hi, I'm under the impression cupy's texture memory is broken.

I attached some code that's working with my previous environment but not the one with latest version of cupy.

### To Reproduce

Code from #7533

```py
import cupy as cp
from cupy.cuda import runtime
from cupy.cuda.texture import (ChannelFormatDescriptor, CUDAarray,
                               ResourceDescriptor, TextureDescriptor,
                               TextureObject,)

def create_texture() -> TextureObject:
    width, height, depth = (64, 32, 19)
    dim = 3 if depth != 0 else 2 if height != 0 else 1

    # generate input data and allocate output buffer
    shape = (depth, height, width) if dim == 3 else \
            (height, width) if dim == 2 else \
            (width,)

    # prepare input, output, and texture memory
    # self.data holds the data stored in the texture memory
    tex_data = cp.random.random(shape, dtype=cp.float32)
    ch = ChannelFormatDescriptor(32, 0, 0, 0,
                                 runtime.cudaChannelFormatKindFloat)
    arr = CUDAarray(ch, width, height, depth)
    arr.copy_from(tex_data)

    # create resource and texture descriptors
    res = ResourceDescriptor(runtime.cudaResourceTypeArray, cuArr=arr)
    address_mode = (runtime.cudaAddressModeClamp,
                    runtime.cudaAddressModeClamp)
    tex = TextureDescriptor(address_mode, runtime.cudaFilterModePoint,
                            runtime.cudaReadModeElementType)

    # create a texture object
    return TextureObject(res, tex)


print("creating texture...")
create_texture()
print("done!")
```

It works with my previous environment but the new one it doesn't.
I tried updating to the latest cupy to check if that was fixed,  but it isn't working.

### Installation

None

### Environment

Doesn't work:
```
OS                           : Linux-5.15.0-88-generic-x86_64-with-glibc2.31
Python Version               : 3.10.10
CuPy Version                 : 13.0.0a1
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.23.5
SciPy Version                : 1.10.1
Cython Build Version         : 0.29.32
Cython Runtime Version       : None
CUDA Root                    : /usr/local/cuda-11.2
nvcc PATH                    : /usr/local/cuda-11.2/bin/nvcc
CUDA Build Version           : 11080
CUDA Driver Version          : 12000
CUDA Runtime Version         : 11070
cuBLAS Version               : (available)
cuFFT Version                : 10702
cuRAND Version               : 10210
cuSOLVER Version             : (11, 3, 5)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 7)
Thrust Version               : 101501
CUB Build Version            : 101501
Jitify Build Version         : 4a37de0
cuDNN Build Version          : 8801
cuDNN Version                : 8201
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce RTX 3090
Device 0 Compute Capability  : 86
Device 0 PCI Bus ID          : 0000:03:00.0
Device 1 Name                : NVIDIA GeForce RTX 3090
Device 1 Compute Capability  : 86
Device 1 PCI Bus ID          : 0000:04:00.0
```

Works:
```
OS                           : Linux-5.15.0-88-generic-x86_64-with-glibc2.31
Python Version               : 3.9.13
CuPy Version                 : 11.5.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.23.3
SciPy Version                : 1.9.1
Cython Build Version         : 0.29.33
Cython Runtime Version       : None
CUDA Root                    : /home/jordao/miniconda3/envs/dexp
nvcc PATH                    : None
CUDA Build Version           : 11020
CUDA Driver Version          : 12000
CUDA Runtime Version         : 11080
cuBLAS Version               : (available)
cuFFT Version                : 10900
cuRAND Version               : 10300
cuSOLVER Version             : (11, 4, 1)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 8)
Thrust Version               : 101000
CUB Build Version            : 101000
Jitify Build Version         : b8d229d
cuDNN Build Version          : 8401
cuDNN Version                : 8401
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : 10500
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce RTX 3090
Device 0 Compute Capability  : 86
Device 0 PCI Bus ID          : 0000:03:00.0
Device 1 Name                : NVIDIA GeForce RTX 3090
Device 1 Compute Capability  : 86
Device 1 PCI Bus ID          : 0000:04:00.0
```

**Comment by kmaehashi** [comment]
Hi @JoOkuma, thanks for reporting! I could reproduce the error with CUDA 11.7 or earlier.

```
creating texture...
Traceback (most recent call last):
  File "/home/maehashi/local/tmp/texture/test.py", line 36, in <module>
    create_texture()
  File "/home/maehashi/local/tmp/texture/test.py", line 32, in create_texture
    return TextureObject(res, tex)
  File "cupy/cuda/texture.pyx", line 531, in cupy.cuda.texture.TextureObject.__init__
  File "cupy_backends/cuda/api/runtime.pyx", line 1011, in cupy_backends.cuda.api.runtime.createTextureObject
  File "cupy_backends/cuda/api/runtime.pyx", line 144, in cupy_backends.cuda.api.runtime.check_status
cupy_backends.cuda.api.runtime.CUDARuntimeError: cudaErrorInvalidValue: invalid argument
```

The problem has gone with CUDA 11.8. Could you try under CUDA 11.8 environment? I'll continue investigating the root cause.

**Comment by leofang** [comment]
No need to try, @kmaehashi, there's an ABI breaking change in cudart texture (around 11.6/11.7, I can find out the bug info if needed) that was fixed in CUDA 11.8. You already gave the correct answer. The cudart version should avoid 11.6/11.7 as far as texture is concerned.

**Comment by JoOkuma** [comment]
@leofang and @kmaehashi, thanks for the tip
